commands/run.go

```diff
@@ -54,11 +65,51 @@ var RunCmd = cli.Command{
 			return node.RunCmd(context.Background(), nil, args...)
 		}
 
+		// Create a list of list of the specified metrics by the user
+		var metricsBefore [][]string
+		if len(collectFlag) != 0 {
+			for _, metric := range collectFlag {
+				// Calculate the values of the metric before the command execution
+				tmp, err := collectMetric(nodes, metric)
+				if err != nil {
+					return err
+				}
+				metricsBefore = append(metricsBefore, tmp)
+			}
+		}
+
 		results, err := mapWithOutput(list, nodes, runCmd)
 		if err != nil {
 			return err
 		}
 
-		return buildReport(results)
+		if len(collectFlag) != 0 {
+			for i, metric := range collectFlag {
+				// Calculate the values of the metric after the command execution
+				tmpMetricsAfter, err := collectMetric(nodes, metric)
+				if err != nil {
+					return err
+				}
+				buildMetricStats(metricsBefore[i], tmpMetricsAfter, metric)
+			}
+		}
+		return buildReport(results, strings.Join(args[:], " "), flagStats)
 	},
 }
+
+func collectMetric(nodes []testbedi.Core, metric string) ([]string, error) {
+	nodeMetricValue := make([]string, len(nodes))
+	for i, node := range nodes {
+		metricNode, ok := node.(testbedi.Metric)
+		if !ok {
+			return nil, fmt.Errorf("node %d does not implement metrics", i)
+		}
+		value, err := metricNode.Metric(metric)
+		if err != nil {
+			return nil, err
+		}
+
+		nodeMetricValue[i] = value
+	}
+	return nodeMetricValue, nil
+}
```

Comment:
Here I opted to collect the metric before and after the command is executed in `run.go`. I think with this approach the code is cleaner and makes more sense.
I could add it in the `mapWithOutput` function but since this function is using locks the more operations I do inside the locks the more the overhead.

I would love to hear your thoughts on which design you think is more useful for the metrics you plan to implement.